components/dialog/RotateEncryptionKeyDialog.vue

```diff
@@ -26,25 +28,31 @@ export default {
   async fetch() {
     const c = this.cluster;
 
-    if ( c.isRke1 && this.$store.getters['isRancher'] ) {
-      let etcdBackups = await this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
+    let etcdBackups = [];
```

Comment:
We may create a function `getEtcdBackups()` and avoid the whole statement.
E.g. (not sure  how it will prettified :stuck_out_tongue:  ):

```ts
const getEtcdBackups = (type, id) => (
    await this.$store.dispatch('rancher/findAll', { type }
  ).filter(backup => backup.clusterId === id);

// not so sure if is mandatory to have at least one of them in these case or elsewise consider it a bug?
const etcdBackups = c.isRke1
  ? getEtcdBackups(NORMAN.ETCD_BACKUP, c.metadata.name)
  : getEtcdBackups(SNAPSHOT, c.id)
```

Reply:
I will create a function named `getEtcdBackups` to better communicate the intent. It's not mandatory to have at least one snapshot. The default should just be an empty array.